mailbox/backup/src/main/java/org/apache/james/mailbox/backup/Archive.java

```diff
@@ -32,5 +32,5 @@
      * @param messages    a stream of MailboxMessages that will be consumed
      * @param destination an OutputStream in which the zip will be written
      */
-    void archive(List<MailboxWithAnnotations> mailboxes, Stream<MailboxMessage> messages, OutputStream destination) throws IOException;
+    void archive(List<MailboxWithAnnotations> mailboxes, Stream<MessageResult> messages, OutputStream destination) throws IOException;
```

Comment:
I'm not sure about this. Did you discussed with other people in Lyon about these changes and  the possibility to rely on Mailbox/Message Mapper ?

Reply:
Yes, mainly with Raphael. I even started this way. 
But the mailbox api is quite closed, and the mapper is an internal component. 
I still discover the api, so i you have others insights on this subject, i'm interested !